machetli/pddl/visitors.py

```diff
@@ -310,3 +310,134 @@ def visit_condition_atom(self, atom) -> Atom:
 
     def visit_condition_negated_atom(self, negated_atom) -> NegatedAtom:
         return Falsity() if contains(negated_atom, self.object_name) else negated_atom
+
+
+class TaskElementEraseInitFactVisitor(TaskElementVisitor):
+    """Partial implementation of TaskElementVisitor interface for deletion of fact from init."""
+
+    def __init__(self, fact):
+        self.the_fact = fact
+
+    def visit_task(self, task):
+        new_init = [atom for atom in task.init if isinstance(atom, Assign) or atom != self.the_fact]
```

Comment:
What is the relevance of the `isinstance(atom, Assign)` check? Is this for PDDL functions?